lib/1.4/dml-builtins.dml

```diff
@@ -36,18 +35,23 @@ constant unmapped_offset = 0xffff_ffff_ffff_ffff;
 
 param NULL auto;
 
-// Handy macros
+// Handy macros and types
+extern int strcmp(const char *__s1, const char *__s2);
+extern void *memset(void *__s, int __c, size_t __n);
+extern typedef uint64_t uintptr_t;
```

Comment:
deserves a test,
local uintptr_t *y;
local uint64_t *x = y;

also you chose different defs for 1.2 and 1.4

Reply:
> deserves a test

I honestly don't get that. _Why?_ It's like if you have a definition `param p = 5;` and then saying "you should have a test that p == 5`. It's pointless. It serves no purpose. If this definition were ever to be changed, it would be due to an intentional choice.

> also you chose different defs for 1.2 and 1.4

_That's_ true, and an oversight on my part. I'll address it.

Reply:
The suggested test is needed because it validates that C agrees with your assumption (uintptr_t and uint64_t are equivalent on all current platforms). It also serves as a reminder to fix the definition on any future platforms where it doesn't hold.

Reply:
> The suggested test is needed because it validates that C agrees with your assumption
Ah, right; the test is meaningful because it's an **extern** typedef. True.

Reply:
Done